test/infrastructure/docker/internal/controllers/dockermachine_controller.go

```diff
@@ -192,6 +185,13 @@ func patchDockerMachine(ctx context.Context, patchHelper *patch.Helper, dockerMa
 func (r *DockerMachineReconciler) reconcileNormal(ctx context.Context, cluster *clusterv1.Cluster, machine *clusterv1.Machine, dockerMachine *infrav1.DockerMachine, externalMachine *docker.Machine, externalLoadBalancer *docker.LoadBalancer) (res ctrl.Result, retErr error) {
 	log := ctrl.LoggerFrom(ctx)
 
+	// Check if the infrastructure is ready, otherwise return and wait for the cluster object to be updated
```

Comment:
Should we document this in the migration doc?

Do we have similar code in CABPK or KCP?

Reply:
This PR already addresses similar case in KCP. I will do a pass on all of your controllers to check if we have similar blocks in other controllers.

Reply:
Ups yeah of course KCP is already addressed :)

Reply:
>Should we document this in the migration doc?

On further thought,  this behavior is very internal to CAPI and does not affect how providers would use CAPI, correct?
Is the idea to highlight that CAPI does not block delete operations on KCP and DockerMachine if Infrastructure is not ready? (a point in the `other` section of the 1.1->1.2 doc?)

Reply:
My thought was. If this applies to our infra provider (CAPD), it's probably something that applies to other infra providers too.

Similar for bootstrap & control plane providers.

Reply:
+1 to get some documentation for providers in case this is missed. Instead of the migration doc (or in addition to), we should have this note in the book section for provider implementers

Reply:
I'm not sure if we currently have documentation on that level e.g. how to implement a infra machine/cluster reconciler (but maybe we have)

Reply:
I think the most detailed doc is: https://cluster-api.sigs.k8s.io/developer/providers/implementers-guide/controllers_and_reconciliation.html
But that doesn't even come close to just looking at KCP / CABPK / CAPD

I don't think we can keep a detailed documentation reasonably up-to-date over time

Reply:
This seemed to be the best place to add the note: https://github.com/kubernetes-sigs/cluster-api/pull/6183/files#diff-1b5f4c4ac9a4463df3857e86252b263cabf6207cf4844a9814f20f698b510d2eR116-R117

Reply:
That link doesn't work for me (GitHub UI just doesn't do anything when I click on it)

Reply:
https://github.com/ykakarap/cluster-api/blob/kcp-machinetemplate-cleanup/docs/book/src/developer/providers/machine-infrastructure.md?plain=1#L116-L117

Reply:
Looks like the perfect place, agree